value/src/main/java/com/google/auto/value/processor/autovalue.vm

```diff
@@ -185,4 +185,105 @@ final class $subclass$formalTypes extends $origClass$actualTypes {
 #if (!$serialVersionUID.empty)
   private static final long serialVersionUID = $serialVersionUID;
 #end
+
+#if ($builderTypeName != "")
+
+  #foreach ($m in $toBuilderMethods)
+
+  @Override
+  public ${builderTypeName}${builderActualTypes} ${m}() {
+    return new Builder${builderActualTypes}(this);
+  }
+
+  #end
+
+  static final class Builder${builderFormalTypes} ##
```

Comment:
I'm not sure if you're referring to an earlier version where we would define a static builder() method in AutoValue_Foo? We don't do that any more: the Foo class is expected to do new AutoValue_Foo.Builder() directly.